packages/style-engine/class-wp-style-engine.php

```diff
@@ -507,19 +607,91 @@ protected static function get_individual_property_css_declarations( $style_value
+ *     'declarations'  => (array) An array of property/value pairs representing parsed CSS declarations.
+ *     'classnames'    => (string) Classnames separated by a space.
  * );
  */
 function wp_style_engine_get_block_supports_styles( $block_styles, $options = array() ) {
-	if ( class_exists( 'WP_Style_Engine' ) ) {
-		$style_engine = WP_Style_Engine::get_instance();
-		return $style_engine->get_block_supports_styles( $block_styles, $options );
+	if ( ! class_exists( 'WP_Style_Engine' ) ) {
+		return array();
+	}
+	$defaults = array(
+		'selector'                   => null,
+		'convert_vars_to_classnames' => false,
+		'enqueue'                    => false,
+	);
+
+	$options       = wp_parse_args( $options, $defaults );
+	$style_engine  = WP_Style_Engine::get_instance();
+	$parsed_styles = $style_engine->parse_block_supports_styles( $block_styles, $options );
+
+	// Output.
+	$styles_output = array();
+	if ( ! empty( $parsed_styles['css_declarations'] ) ) {
+		$styles_output['css']          = $style_engine->compile_css( $parsed_styles['css_declarations'], $options['selector'] );
+		$styles_output['declarations'] = $parsed_styles['css_declarations'];
+		if ( true === $options['enqueue'] ) {
+			$style_engine::store_css_rule( $options['selector'], $parsed_styles['css_declarations'], 'block-supports' );
+		}
+	}
+
+	if ( ! empty( $parsed_styles['classnames'] ) ) {
+		$styles_output['classnames'] = implode( ' ', array_unique( $parsed_styles['classnames'] ) );
+	}
+
+	return array_filter( $styles_output );
+}
+
+/**
+ * Global public interface method to register styles to be enqueued and rendered.
+ *
+ * @access public
+ *
+ * @param string $store_key A valid store key.
+ * @param array  $css_rules array(
+ *     'selector'         => (string) A CSS selector.
+ *     'css_declarations' => (boolean) An array of CSS definitions, e.g., array( "$property" => "$value" ).
+ * );.
+ *
+ * @return WP_Style_Engine_CSS_Rules_Store.
+ */
+function wp_style_engine_add_to_store( $store_key, $css_rules = array() ) {
+	$style_engine = WP_Style_Engine::get_instance();
+	if ( empty( $css_rules ) ) {
+		return $style_engine::get_store( $store_key );
+	}
+
+	foreach ( $css_rules as $selector => $css_declarations ) {
+		if ( empty( $selector ) || empty( $css_declarations ) ) {
+			continue;
+		}
+		$style_engine::store_css_rule( $selector, $css_declarations, $store_key );
+	}
+	return $style_engine::get_store( $store_key );
+}
+
+/**
+ * Returns a compiled stylesheet from stored CSS rules.
+ *
+ * @access public
+ *
+ * @param string $store_key        A valid store key.
+ *
+ * @return string A compiled stylesheet from stored CSS rules.
+ */
+function wp_style_engine_get_stylesheet( $store_key ) {
+	if ( empty( $store_key ) ) {
+		return '';
 	}
-	return null;
+	return WP_Style_Engine::get_instance()::compile_stylesheet_from_store( $store_key );
```

Comment:
I believe we don't need to get the instance here, right?

```suggestion
	return WP_Style_Engine::compile_stylesheet_from_store( $store_key );
```